packages/teams-js/src/public/media.ts

```diff
@@ -412,14 +412,17 @@ export namespace media {
      * @param mediaEvent indicates what the event that needs to be signaled to the host client
      * Optional; @param callback is used to send app if host client has successfully handled the notification event or not
      */
-    protected notifyEventToHost(mediaEvent: MediaControllerEvent, callback?: (err?: SdkError) => void): void {
+    protected notifyEventToHost(
```

Comment:
I'm going to look at this more tomorrow, but I was expecting to see this function and the VideoController.stop function (so the MediaController.stop function as well) be converted to return Promises, like `selectMedia` was. With overloads to support the existing callback pattern that are marked deprecated and a new function that returns a Promise. Will provide more detail tomorrow!

Reply:
I agree with Erin. The ultimate goal of this change should be to make the public, user facing function of MediaController.stop() return a Promise and throw an exception if an error occurs. In order to continue supporting back compat with v1 we need to have multiple signatures for stop (which is a pattern they follow in other places in this code base. As Erin pointed out, selectMedia is one example of this pattern.

I think the signature we'd like to add for stop is something like:

public stop(): Promise<void>

And then notifyEventToHost and isApiSupportedOnMobile should both throw SdkErrors instead of returning them

Happy to talk more if it would be helpful!